tools/scaffolder/README.md

```diff
@@ -0,0 +1,139 @@
+# Atlas Controller Scaffolder
+
+A tool to generate Kubernetes controllers for MongoDB Atlas resources based on CRD configurations.
+
+> [!WARNING]
+> This is the experimental tool. Bugs and issues are still present
+
+## Overview
+
+This scaffolder generates Kubernetes controllers that follow the MongoDB Atlas Kubernetes operator patterns, including:
+
+- **State machine-based controllers** with proper lifecycle management
+- **Translation layers** for Atlas SDK integration
+- **Service interfaces** with appropriate Atlas API mappings
+- **License headers** and proper package structure
+
+## Dependencies
+
+### Required Local Repositories
+
+The following repositories must be available locally in the parent directory:
+
+```
+work/
+├── atlas-controller-scaffolder/    # This tool
+├── atlas2crd/                      # Config definitions
+└── mongodb-atlas-kubernetes/       # Target operator
+```
+
+### Setup
+
+1. **Clone dependencies:**
+
+   ```bash
+   cd ../
+   git clone https://github.com/mongodb-atlas-kubernetes/atlas2crd
+   git clone https://github.com/mongodb/mongodb-atlas-kubernetes
+   ```
+
+2. **Replace the crd2go dependency**
+   Make sure the crd2go dependency is pointing to the local copy. In your `go.mod` file:
+
+   ```bash
+   replace github.com/josvazg/crd2go => ../crd2go
+   ```
```

Comment:
This will be just be a dependency on the open source repo, once available. It can now already be used a a library, once the deps got fixed.